### ISSUE TITLE
Bannerlord d3d_device issue on campaign start

### ISSUE THREAD

**Issue by DuckDuckG0053** [italomandara/CXPatcher#163]
I tried running Bannerlord and everything seems ok until loading into the campaign after character creation. The error is "d3d_device CreateShaderResourceView failed at rglGPU_device create_gpu_buffer: invalid parameter". How can i fix this error please?

**Comment by DuckDuckG0053** [comment]
That's the error when running with DXVK, when running with D3DMetal, there is an error immediately on start "Cannot create GPU device!"

**Comment by italomandara** [comment]
Try by redownloading Crossover.app, Latest CXPatcher for that version of crossover, create a new bottle and install your games/steam there.

What's your hardware?
What's your os?
What version of CXPatcher?
What Version of crossover?
are you using steam, origin or other methods?

**Comment by oceaniver** [comment]
Crossover 24,mac os sonama,steam,the same problem

**Comment by italomandara** [comment]
https://www.protondb.com/app/261550#lvurrxVftA

many users are reporting that the game doesn't work if steam overlay is on

**Comment by alver1584** [comment]
It is not about steam overlay as I see. I have a ticket is the Crossover support, but they don't answer. You can vote for support of Bannerlord maybe it will make the difference... https://www.codeweavers.com/compatibility/crossover/mount-and-blade-ii-bannerlord

I have no idea about all technical details but it seems to be the problem with directX. I tried to use other d3d libraries in wine configuration but didn't succeed. Maybe it will be fixed in next updates, there is developing version of wine 9.5 already. But without miracle I didn't think that solution can be found in the nearest future :(( would be really happy to be wrong about that

**Comment by alver1584** [comment]
tried It with V0.5.0-pre for Crossover 24, but nothing changes. (don't know what did I expect)

**Comment by italomandara** [comment]
> That's the error when running with DXVK, when running with D3DMetal, there is an error immediately on start "Cannot create GPU device!"

have you tried with both dxvk and d3dmetal turned off? It's slower but it could work

**Comment by alver1584** [comment]
> have you tried with both dxvk and d3dmetal turned off? It's slower but it could work

<img width="373" alt="Снимок экрана 2024-04-01 в 12 07 47" src="https://github.com/italomandara/CXPatcher/assets/162310247/23c13c67-1cf2-4c6f-91b8-043b06354c7f">

Turned both of them off

**Comment by alver1584** [comment]
I believe it could be relate with this wine bug: https://bugs.winehq.org/show_bug.cgi?id=48804

Any chance to fix it with CXpatcher?

**Comment by italomandara** [comment]
What's your hardware?

**Comment by alver1584** [comment]
> What's your hardware?

MacBook Air m2, Sonoma, crossover 24.0.1

**Comment by alver1584** [comment]
I can provide a log if you are interested. I am not a developer, but tried everything that I could to make this game run 
[backtrace.txt](https://github.com/italomandara/CXPatcher/files/14898092/backtrace.txt)

**Comment by italomandara** [comment]
> I can provide a log if you are interested. I am not a developer, but tried everything that I could to make this game run [backtrace.txt](https://github.com/italomandara/CXPatcher/files/14898092/backtrace.txt)

On crossover there are no guarantees a game will run, some games just won't, if you provide some logs I could look into it but it could be something not fixable via cxpatcher.

**Comment by alver1584** [comment]
> On crossover there are no guarantees a game will run, some games just won't, if you provide some logs I could look into it but it could be something not fixable via cxpatcher.

Yeah, I understand, but i kind of obsessed with the idea to play Bannerlord, so really appreciate your help! 

So here is a backtrace file is it enough? If not, could you advise me how to get a proper log? 
[backtrace123.txt](https://github.com/italomandara/CXPatcher/files/14905392/backtrace123.txt)

**Comment by italomandara** [comment]
> > On crossover there are no guarantees a game will run, some games just won't, if you provide some logs I could look into it but it could be something not fixable via cxpatcher.
> 
> Yeah, I understand, but i kind of obsessed with the idea to play Bannerlord, so really appreciate your help!
> 
> So here is a backtrace file is it enough? If not, could you advise me how to get a proper log? [backtrace123.txt](https://github.com/italomandara/CXPatcher/files/14905392/backtrace123.txt)

This doesn't help much you need to run steam or your game by right clicking -> run with options and enable logs.

**Comment by alver1584** [comment]
> > > On crossover there are no guarantees a game will run, some games just won't, if you provide some logs I could look into it but it could be something not fixable via cxpatcher.
> > 
> > 
> > Yeah, I understand, but i kind of obsessed with the idea to play Bannerlord, so really appreciate your help!
> > So here is a backtrace file is it enough? If not, could you advise me how to get a proper log? [backtrace123.txt](https://github.com/italomandara/CXPatcher/files/14905392/backtrace123.txt)
> 
> This doesn't help much you need to run steam or your game by right clicking -> run with options and enable logs.

Oh, okay! That should be the right one then (made zip-archive to send) 

[Bannerlord.Native 2.cxlog.zip](https://github.com/italomandara/CXPatcher/files/14906028/Bannerlord.Native.2.cxlog.zip)

**Comment by italomandara** [comment]
the only thing i can see is:
D3D11RasterizerState: Antialiased lines not supported

It seems like Gothic 2 has the same issue with dxvk
if you're an expert you could try this workaround:
https://github.com/kirides/GD3D11/issues/14#issuecomment-739528956
but I doubt it works.

**Comment by AndersAskeland** [comment]
To add some info, one can now actually get in game by using DXMT, however, when playing in a 3D view, there is a lot of texture flickering.

**Comment by italomandara** [comment]
> To add some info, one can now actually get in game by using DXMT, however, when playing in "walk around mode", there is a lot of texture flickering.
> 
> ![Image](https://github.com/user-attachments/assets/1cd98229-f93b-4ddb-b9f0-ee4cef2ddc57)

This could be an interesting thing to pass to the author of dxmt, he's pretty busy debuggning other stuff but you could opena a report and help him debug.

**Comment by le0nard01** [comment]
With the new version of DXMT (v0.41), it's working: https://github.com/3Shain/dxmt/releases/tag/v0.41

<img width="1160" alt="Image" src="https://github.com/user-attachments/assets/1bbf1b3a-95d6-421b-b429-6c863a2a0466" />
